packages/superset-ui-dimension/src/getMultipleTextDimensions.ts

```diff
@@ -0,0 +1,61 @@
+import { TextStyle, Dimension } from './types';
+import getBBoxCeil from './svg/getBBoxCeil';
+import { hiddenSvgFactory, textFactory } from './svg/factories';
+import updateTextNode from './svg/updateTextNode';
+
+/**
+ * get dimensions of multiple texts with same style
+ * @param input
+ * @param defaultDimension
+ */
+export default function getMultipleTextDimensions(
+  input: {
+    className?: string;
+    container?: HTMLElement;
+    style?: TextStyle;
+    texts: string[];
+  },
+  defaultDimension?: Dimension,
+): Dimension[] {
+  const { texts, className, style, container } = input;
+
+  const cache = new Map<string, Dimension>();
+  let textNode: SVGTextElement | undefined;
+  let svgNode: SVGSVGElement | undefined;
+
+  const dimensions = texts.map(text => {
+    // Empty string
+    if (text.length === 0) {
+      return { height: 0, width: 0 };
```

Comment:
instead of this special case, why not initialize the cache with:
```
{ "": { height: 0, width: 0 } }
```

then we don't need to check both cases every time (and i'd imagine the O(1) cache lookup would be about the same speed as the string length compare)

Reply:
I generally agree with this, but the [W3C SVGGraphicsElement spec](https://svgwg.org/svg2-draft/types.html#InterfaceSVGGraphicsElement) has a method named getBBox (that we use on line 9), so this instance may be an acceptable exception